### PR TITLE
chore: [IOBP-319] Add new wallet payment redux and saga

### PR DESCRIPTION
**Pull request by mastro993**
## Short description
This PR introduces the reducer and networking sagas to handle the new wallet payment flow.
The flow covers only the happy path and displays the store content in the screens.
**Note:** the authorization process is not covered by this implementation and will be handled in [IOBP-432](https://pagopa.atlassian.net/browse/IOBP-432)

https://github.com/pagopa/io-app/assets/6160324/201e42e4-13ad-465c-a5a3-8bd517c11cb2

## List of changes proposed in this pull request
- Added Wallet eCommerce API definitons url and script in `package.json`
- Added `DebugPrettyPrint`, an utility component to display the content of any object in a pleasant and readable way.
- Renamed `WALLET_API_BASEURL` and `WALLET_API_UAT_BASEURL` in `.env.local`, `.env.production` and `config.ts` files
- Moved barcode scan files from `ts/features/walletV3/payment` into dedicated subfolder in `ts/features/walletV3/barcode`
- Added client, saga and reducer for Wallet eCommerce
- Added barebone logic into wallet payment flow

## How to test
Go to **Profile > New Wallet Playground > Payment Playground** using the `io-dev-api-server` as the backend. Just make sure to checkout this branch: https://github.com/pagopa/io-dev-api-server/pull/322 and set the following in the config.json

```json
  "wallet": {
    "useLegacyRptIdVerificationSystem": true
  }
```

[IOBP-432]: https://pagopa.atlassian.net/browse/IOBP-432?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: ts/features/walletV3/payment/store/reducers/index.ts [hunk 1]
```diff
@@ -0,0 +1,174 @@
+import * as pot from "@pagopa/ts-commons/lib/pot";
```
Comment (forrest57) [inline ts/features/walletV3/payment/store/reducers/index.ts:1]: could it be a good idea to divide this reducer into multiple files?

File: ts/navigation/AuthenticatedStackNavigator.tsx [hunk 2]
```diff
@@ -285,13 +286,9 @@ const AuthenticatedStackNavigator = () => {
...
         }}
       />
-      {/* 
-        This screen is outside the WalletPaymentNavigator to enable the slide from bottom animation.
-        FIXME IOBP-383: Using react-navigation 6.x we can achive this using a Stack.Group inside the WalletPaymentNavigator
-      */}
       <Stack.Screen
-        name={WalletPaymentRoutes.WALLET_PAYMENT_BARCODE_SCAN}
-        component={WalletPaymentBarcodeScanScreen}
+        name={WalletBarcodeRoutes.WALLET_BARCODE_MAIN}
```
Comment (forrest57) [inline ts/navigation/AuthenticatedStackNavigator.tsx:290]: why remove the comment? unless the issue is not relevant anymore, ofc

File: ts/features/walletV3/payment/store/reducers/index.ts (hunk 1 above)
Comment (mastro993) [reply ts/features/walletV3/payment/store/reducers/index.ts:1]: In that case you need to split the reducer in multiple reducers. I don't think it's a good idea.

File: ts/navigation/AuthenticatedStackNavigator.tsx (hunk 2 above)
Comment (mastro993) [reply ts/navigation/AuthenticatedStackNavigator.tsx:290]: The issue is not relevant anymore :)

File: ts/navigation/params/AppParamsList.ts [hunk 3]
```diff
@@ -105,7 +107,7 @@ export type AppParamsList = {
 
   [WalletOnboardingRoutes.WALLET_ONBOARDING_MAIN]: NavigatorScreenParams<WalletOnboardingParamsList>;
   [WalletPaymentRoutes.WALLET_PAYMENT_MAIN]: NavigatorScreenParams<WalletPaymentParamsList>;
-  [WalletPaymentRoutes.WALLET_PAYMENT_BARCODE_SCAN]: undefined; // FIXME IOBP-383: remove after react-navigation 6.x upgrade. This should be insde WALLET_PAYMENT_MAIN
+  [WalletBarcodeRoutes.WALLET_BARCODE_MAIN]: NavigatorScreenParams<WalletBarcodeParamsList>;
```
[the remark this answers is not in the document]
Comment (mastro993) [reply ts/navigation/params/AppParamsList.ts:110]: Same!

**Comment by forrest57** [comment]
The testing case you provided breaks as of now, with a generic `potNone`; I've tried with the dev-server both in master and in `IOBP-404`.
could you check?

**Comment by mastro993** [comment]
> The testing case you provided breaks as of now, with a generic `potNone`; I've tried with the dev-server both in master and in `IOBP-404`. could you check?

The problem is that the payment does not exists.
I forgot to mention that in the config.json you must add the following:

```
  "wallet": {
    "useLegacyRptIdVerificationSystem": true
  },
  ```

**Review by forrest57 — APPROVED** [review]

**Pull request merged**